src/metricCollector.ts

```diff
@@ -59,14 +59,16 @@ export class MetricCollector {
       }
       this.logger.info('added queue', name);
       this.queuesByName.set(name, {
-        name,
-        queue: new Queue(name, {
-          ...this.bullOpts,
-          connection: this.defaultRedisClient,
-        }),
-        prefix: this.bullOpts.prefix || 'bull',
-        queueEvents: new QueueEvents(name)
-      });
+				name,
+				queue: new Queue(name, {
+					...this.bullOpts,
+					connection: this.defaultRedisClient,
+				}),
+				prefix: this.bullOpts.prefix || 'bull',
+				queueEvents: new QueueEvents(name, {
+					connection: this.defaultRedisClient,
```

Comment:
it seems we cannot reuse connection between Queue and QueueEvents instances
> Queue and Worker classes can accept an existing ioredis instance, and by that reusing that connection, however QueueScheduler and QueueEvents cannot do that because they require blocking connections to Redis
https://docs.bullmq.io/guide/connections

so plz consider using a separate one 🙏